coordinator/src/main/scala/filodb.coordinator/queryplanner/DefaultPlanner.scala

```diff
@@ -338,17 +342,24 @@ trait  DefaultPlanner {
                                         sqww: SubqueryWithWindowing
                                       ) : PlanResult = {
     // absent over time is essentially sum(last(series)) sent through AbsentFunctionMapper
-    innerExecPlan.plans.foreach(
-      _.addRangeVectorTransformer(PeriodicSamplesMapper(
-        sqww.startMs, sqww.stepMs, sqww.endMs,
-        window,
-        Some(InternalRangeFunction.lpToInternalFunc(RangeFunctionId.Last)),
-        qContext,
-        stepMultipleNotationUsed = false,
-        Seq(),
-        offsetMs,
-        rawSource = false
-      ))
+     val realStartMs = sqww.atMs.getOrElse(sqww.startMs)
+     val realEndMs = sqww.atMs.getOrElse(sqww.endMs)
+     val realStep = sqww.atMs.map(_ => 0L).getOrElse(sqww.stepMs)
+
+     innerExecPlan.plans.foreach(plan => {
+       plan.addRangeVectorTransformer(PeriodicSamplesMapper(
+         realStartMs, realStep, realEndMs,
+         window,
+         Some(InternalRangeFunction.lpToInternalFunc(RangeFunctionId.Last)),
+         qContext,
+         stepMultipleNotationUsed = false,
+         Seq(),
+         offsetMs,
+         rawSource = false
+       ))
+       sqww.atMs.map(_ => plan.addRangeVectorTransformer(RepeatTransformer(sqww.startMs, sqww.stepMs, sqww.endMs,
```

Comment:
Similar question to the above: shouldn't the `RepeatTransformer` be passed the `real` time params?

Reply:
The `RepeatTransformer` should take the original parameters.
Say we have range query` [1, 10, 100] @ 50`
We will have the following plan.
```
--RepeatTransformer(1, 10, 100) => populate the value returned by PeriodicMapper  forever every step
   --PeriodicMapper(50,0,50) => return only one value at timestamp 50
```

Reply:
sorry. I think maybe the term real is confusing.
the real time is the time with modifier applied.
Maybe I should change realStartMs to something like dataStartMs.